website/src/app/pages/data-distributions/charts/bar-chart.vl.ts

```diff
@@ -0,0 +1,213 @@
+
+export function createBarSpec(variable: VariableData, distributionData: DistributionData[] = []): VisualizationSpec {
+  return {
+    $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
+    datasets: {
+      distribution: distributionData
+    },
+    height: 300,
+    data: {
+      name: 'distribution'
+    },
+    view: {
+      strokeOpacity: 0
+    },
+    config: {
+      concat: {
+        spacing: 200,
+        columns: 2
+      }
+    },
+    hconcat: [
+      {
+        title: {
+          text: `${variable.dataset} by ${variable.name}`,
+        },
+        view: {
+          strokeOpacity: 0,
+        },
+        layer: [
+          {
+            data: {
+              values: [
+                /* eslint-disable @typescript-eslint/naming-convention */
+                { y: 1.7, LABEL: 'Type:' },
+                { y: 1.6, LABEL: 'Description:' },
+                { y: 1.5, LABEL: 'Missing values:' }
+                /* eslint-enable @typescript-eslint/naming-convention */
+              ]
+            },
+            mark: {
+              type: 'text',
+              align: 'left',
+              fontWeight: 'bold',
+              xOffset: -50
+            },
+            encoding: {
+              text: {
+                type: 'nominal',
+                field: 'LABEL'
+              },
+              y: {
+                type: 'quantitative',
+                field: 'y',
+                axis: null
+              },
+              color: {
+                value: '#212121'
+              }
+            }
+          },
+          {
+            data: {
+              values: [
+                /* eslint-disable @typescript-eslint/naming-convention */
+                { y: 1.7, VALUE: `${variable.type}` },
+                { y: 1.6, VALUE: `${variable.description}` },
+                { y: 1.5, VALUE: `${variable.missingValues.toFixed(1)}%` }
+                /* eslint-enable @typescript-eslint/naming-convention */
+              ]
+            },
+            mark: {
+              type: 'text',
+              align: 'left',
+              xOffset: 40
+            },
+            encoding: {
+              text: {
+                type: 'nominal',
+                field: 'VALUE'
+              },
+              y: {
+                type: 'quantitative',
+                field: 'y',
+                axis: null
+              },
+              color: {
+                value: '#212121'
+              }
+            }
+          }
+        ]
+      },
+      {
+        view: {
+          strokeOpacity: 0
+        },
+        width: 500,
+        transform: [
+          {
+            calculate: 'slice(datum.period, 0, 4)',
+            as: 'year'
+          },
+          {
+            aggregate: [{
+              op: 'sum',
+              field: 'count',
+              as: 'total'
+            }],
+            groupby: variable.horizontal ? ['value'] : ['year']
```

Comment:
horizontal shows `value` in the Y axis and `count` in the X. "non-horizontal" shows `count` in the Y axis and `value` in the X. You never have to compute year (nor should you reference period as this will be pre-aggregated for you).